app/decorators/controllers/solidus_bolt/spree_checkout_controller/add_addresses_to_bolt.rb

```diff
@@ -13,6 +15,23 @@ def finalize_order
         super
       end
 
+      private
+
+      def bolt_addresses
+        @bolt_addresses ||= SolidusBolt::Accounts::DetailService.call(
+          access_token: session[:bolt_access_token]
+        )['addresses']
+      end
+
+      def address_already_uploaded?(address)
+        bolt_addresses.any? do |bolt_address|
+          bolt_address['street_address1'] == address.address1 &&
+          bolt_address['locality'] == address.city &&
+          bolt_address['region'] == address.state&.abbr &&
+          bolt_address['postal_code'] == address.zipcode
+        end
+      end
+
```

Comment:
What do you think to move that logic into the service?

Reply:
fixed in https://github.com/nebulab/solidus_bolt/pull/96/commits/97559ed0f23fc53d55fc220a74f77097149c97cb